File: social-network-visualizer-backend/src/main/java/com/example/social_network_visualizer_backend/dto/request/GraphQueryRequest.java
package com.example.social_network_visualizer_backend.dto.request;

import com.example.social_network_visualizer_backend.enums.NodeType;
import com.example.social_network_visualizer_backend.enums.RelationType;
import java.util.Set;

public record GraphQueryRequest(Set<NodeType> nodeTypes, Set<RelationType> relationTypes) {}

File: social-network-visualizer-frontend/src/types/GraphQueryRequest.ts
import { NodeType, RelationType } from "./GraphTypes";

export enum FetchStrategy {
  LIMIT_PER_TYPE = "LIMIT_PER_TYPE",
  ALL = "ALL",
}

export interface FetchConfig {
  strategy: FetchStrategy;
  nodeLimits?: Record<NodeType, number>;
}

export interface GraphQueryRequest {
  nodeTypes: NodeType[];
  relationTypes: RelationType[];
  fetchConfig?: FetchConfig; // FetchStrategy.ALL if undefined
}
